klaus/repo.py

```diff
@@ -58,10 +58,21 @@ def namespaced_name(self):
 
     # TODO: factor out stuff into dulwich
     def get_last_updated_at(self):
-        """Get datetime of last commit to this repository."""
-        # Cache result to speed up repo_list.html template.
-        # If self.get_refs() has changed, we should invalidate the cache.
-        all_refs = self.get_refs()
+        """Get datetime of last commit to this repository.
+
+        Caches the result to speed up the repo_list page.
+        Cache is invalidated if one of the ref targets changes,
+        eg. a new commit has been made and 'refs/heads/master' was changed.
+        """
+        if len(self.refs.keys()) > 10000:
```

Comment:
10000 still seems like quite a lot, there's no way to meaningfully display that - and it still means 10k random accesses of the repository.

What if there are more than 10000 heads, as is the case with the nix repo?

Reply:
Ah damn I thought that most heads are branches but eg. GH PRs are also heads

Reply:
Note that the number used here is not displayed anywhere, just the maximum timestamp of the refs

Reply:
> Ah damn I thought that most heads are branches but eg. GH PRs are also heads

Wait no they are not